google_compute_engine/networking/network_daemon.py

```diff
@@ -0,0 +1,187 @@
+    """Constructor.
+
+    Args:
+      ip_forwarding_enabled: bool, True if ip forwarding is enabled.
+      proto_id: string, the routing protocol identifier for Google IP changes.
+      ip_aliases: bool, True if the guest should configure IP alias routes.
+      target_instance_ips: bool, True supports internal IP load balancing.
+      dhclient_script: string, the path to a dhclient script used by dhclient.
+      dhcp_command: string, a command to enable Ethernet interfaces.
+      dhcp_refresh_enabled: bool, True if dhcp refresh is enabled.
+      network_setup_enabled: bool, True if network setup is enabled.
+      debug: bool, True if debug output should write to the console.
+    """
+    facility = logging.handlers.SysLogHandler.LOG_DAEMON
+    self.logger = logger.Logger(
+        name='google-networking', debug=debug, facility=facility)
+    self.network_utils = network_utils.NetworkUtils(logger=self.logger)
+    self.watcher = metadata_watcher.MetadataWatcher(logger=self.logger)
+
+    self.ip_aliases = ip_aliases
+    self.target_instance_ips = target_instance_ips
+
+    # Get initial metadata.
+    result = self.watcher.GetMetadata(
+        metadata_key=self.network_interfaces, recursive=True)
+    interface_details = self._ExtractInterfaceMetadata(result)
+
+    if network_setup_enabled:
+      interfaces = [interface for interface, _, _ in interface_details]
+      network_setup.NetworkSetup(
+          interfaces, dhclient_script=dhclient_script,
+          dhcp_command=dhcp_command, debug=debug)
+
+    self.ip_forwarding_enabled = ip_forwarding_enabled
+    if ip_forwarding_enabled:
+      self.ip_forwarding = ip_forwarding.IpForwarding(proto_id, debug)
+
+    self.dhcp_refresh_enabled = dhcp_refresh_enabled
+    if dhcp_refresh_enabled:
+      dhcpv6_tokens = {name: dhcpv6_token
+          for name, _, dhcpv6_token in interface_details}
+      self.dhcp_refresh = dhcp_refresh.DhcpRefresh(dhcpv6_tokens, debug)
+
+    try:
+      with file_utils.LockFile(LOCKFILE):
+        self.logger.info('Starting Google Networking daemon.')
+        timeout = 60 + random.randint(0, 30)
+        self.watcher.WatchMetadata(
+            self.HandleNetworkInterfaces, metadata_key=self.network_interfaces,
+            recursive=True, timeout=timeout)
+    except (IOError, OSError) as e:
+      self.logger.warning(str(e))
+
+  def HandleNetworkInterfaces(self, result):
+    """Called when network interface metadata changes.
+
+    Args:
+      result: dict, the metadata response with the network interfaces.
+    """
+    interface_details = self._ExtractInterfaceMetadata(result)
+
+    for interface, forwarded_ips, dhcpv6_refresh_token in interface_details:
+      if self.ip_forwarding_enabled:
+        self.ip_forwarding.HandleForwardedIps(interface, forwarded_ips)
+      if self.dhcp_refresh_enabled:
+        self.dhcp_refresh.RefreshDhcp(interface, dhcpv6_refresh_token)
+
+  def _ExtractInterfaceMetadata(self, metadata):
```

Comment:
I would suggest creating an network interface class that has three properties:
- name
- forwarded_ips
- refresh_token